### PR TITLE
DOC, MAINT: test theme CSS fixes

### PR DESCRIPTION
**Pull request by drammock**
This updates our site CSS and javascript in response to some upstream theme bugfixes:

- https://github.com/pydata/pydata-sphinx-theme/pull/313 (already merged) handles the visibility of the version dropdown / hamburger menu
- https://github.com/pydata/pydata-sphinx-theme/pull/318 (not yet merged) handles the sticky top navbar covering up in-page target anchors.

Once the second PR above is merged, we'll need to update the theme commit that we pin to in `requirements_doc.txt`

### CODE REVIEW

**Comment by wmvanvliet** [comment]
Link to doc build: https://25434-1301584-gh.circle-artifacts.com/0/dev/index.html

**Comment by drammock** [comment]
something is going wrong with CircleCI here.  It is definitely checking out the correct version of the theme:

> Collecting git+git://github.com/pydata/pydata-sphinx-theme.git@93b5115834435c995138da4be9ad7bbcfbca8127 (from -r requirements_doc.txt (line 3))

But in the rendered site, it's using a version of the theme CSS file that does not exist in the checked-out version of the theme (the theme generates the CSS from SCSS sources, and incorporates a hash into the generated CSS filename, so it's really obvious when this has happened).  Specifically, the docs should be using `index.0e366ba14472447708f75ccb7d8f24a3.css` (see [here](https://github.com/pydata/pydata-sphinx-theme/tree/93b5115834435c995138da4be9ad7bbcfbca8127/pydata_sphinx_theme/static/css)) but instead it is using the old css file:

> Uploading /home/circleci/project/doc/_build/html/_static/css/index.951c8c8e3af89de180a2f03968e0e7cb.css (154 kB): DONE

(note the different hash).  I think this might be related to our new-ish procedure of "restoring cache" to save build time?

**Comment by drammock** [comment]
I've just triggered a circle re-run to see if it fixes itself.

**Comment by larsoner** [comment]
CircleCI is great because you can remotely debug it, in the upper right you can "Rebuild with SSH"

**Comment by drammock** [comment]
didn't fix itself; re-running with SSH

**Comment by drammock** [comment]
What I've determined so far:

1. the installed version of the theme (which ends up at `~/.local/lib/python3.8/site-packages/pydata_sphinx_theme`) has a lot of files with modification dates of Feb 12 (today is Feb 26).  This does appear to come from a cache:
```
Found a cache from build 25156 at user-install-bin-cache
Size: 423 MiB
Cached paths:
  * /home/circleci/.local/lib/python3.8/site-packages
  * /home/circleci/.local/bin

Downloading cache archive...
Validating cache...

Unarchiving cache...
```
(from the last "restoring cache" step right before the "get python running" step)

2. During the "get Python running" step, CircleCI clones `pydata_sphinx_theme` to `/tmp/pip-req-build-151r8l7v` (which is gone by the time I went looking for it).  There is also no pip output line saying that it has successfully installed pydata-sphinx-theme; there are lines about cloning and verifying it:
```
Collecting git+git://github.com/pydata/pydata-sphinx-theme.git@93b5115834435c995138da4be9ad7bbcfbca8127 (from -r requirements_doc.txt (line 3))
  Cloning git://github.com/pydata/pydata-sphinx-theme.git (to revision 93b5115834435c995138da4be9ad7bbcfbca8127) to /tmp/pip-req-build-151r8l7v
  Running command git clone -q git://github.com/pydata/pydata-sphinx-theme.git /tmp/pip-req-build-151r8l7v
  Running command git rev-parse -q --verify 'sha^93b5115834435c995138da4be9ad7bbcfbca8127'
  Running command git fetch -q git://github.com/pydata/pydata-sphinx-theme.git 93b5115834435c995138da4be9ad7bbcfbca8127
```
...and there are lines later about finding its dependencies are already present:
```
Requirement already satisfied: beautifulsoup4 in /home/circleci/.local/lib/python3.8/site-packages (from pydata-sphinx-theme==0.4.2.dev0->-r requirements_doc.txt (line 3)) (4.9.3)
...
Requirement already satisfied: soupsieve>1.2 in /home/circleci/.local/lib/python3.8/site-packages (from beautifulsoup4->pydata-sphinx-theme==0.4.2.dev0->-r requirements_doc.txt (line 3)) (2.2)
```
...but there is never a line saying "requirement already satisified: pyvista-sphinx-theme", nor is there a line saying "successfully installed pydata-sphinx-theme".  So AFAICT the updated version is getting cloned but not installed (IDK why) and the only reason this doesn't crash the doc build is that an older version of the theme already exists (restored from cache).

File: requirements_doc.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 sphinx
 https://github.com/numpy/numpydoc/archive/master.zip
-git+git://github.com/pydata/pydata-sphinx-theme.git@270bf6c60de2c9f16aa32c3eeb6de39f0ee0e734
+git+git://github.com/pydata/pydata-sphinx-theme.git@93b5115834435c995138da4be9ad7bbcfbca8127
```
Comment (larsoner) [inline requirements_doc.txt:3]: I would avoid `git+` syntax, it's slower and requires git. I'd try
```suggestion
https://github.com/pydata/pydata-sphinx-theme/zipball/93b5115834435c995138da4be9ad7bbcfbca8127
```
and maybe it automagically fixes things, but even if it doesn't it's probably better

**Comment by larsoner** [comment]
Didn't seem to work, pushed a workaround

**Comment by drammock** [comment]
Docs/CSS look correct now.  Failures are `brain.screenshot()` on mac, and the usual pip-pre failures that we're working on in #8955.  +1 for merge

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**